smart-contracts/assembly/contracts/pool.ts

```diff
@@ -142,7 +132,7 @@ export function addLiquidity(binaryArgs: StaticArray<u8>): void {
   const reserveB = _getLocalReserveB();
 
   // get the total supply of the LP token
-  const totalSupply: u256 = u256.from(liquidityManager.getTotalSupply());
+  const totalSupply: u256 = liquidityManager.getTotalSupply();
```

Comment:
Don't forget to send before the protocol fees from input to the right receiver and to update the amounts in the pool.